afm/filesystems/vault.py

```diff
@@ -34,34 +36,36 @@ def get_raw_secret_from_vault(jwt, secret_path, vault_address, vault_path, role)
         logger.error("Malformed vault authorization response")
         return None
     client_token = vault_auth_response["auth"]["client_token"]
-    logger.debug("client_token: %s", str(client_token))
     secret_full_path = vault_address + secret_path
-    logger.debug("secret_full_path = %s", str(secret_full_path))
     response = requests.get(secret_full_path, headers={"X-Vault-Token" : client_token})
-    logger.error("Status code from Vault response: " + str(response.status_code))
+    logger.debug('vault response',
+        extra={'secret_full_path': str(secret_full_path),
+               StatusCode: str(response.status_code)})
     if response.status_code == 200:
         response_json = response.json()
         if 'data' in response_json:
             return response_json['data']
         else:
             logger.error("Malformed secret response. Expected the 'data' field in JSON")
     else:
-        logger.error("Got error code %d requesting Vault secret", response.status_code)
-        logger.error("Error response: %s", str(response.json()))
+        logger.error("vault error",
+            extra={StatusCode: str(response.status_code),
+                   Error: str(response.json())})
```

Comment:
I would expect the output to include { "error": status_code, "message": "Error reading credentials from vault ", response.json, "DataSetID": relevant ID }.  Not sure that's what's produced here.

Reply:
Addressed?

Reply:
Rather than appending the response json to the message, I think it is better to append it to the "error" field.
What do you think?